src/update-changelog.ts

```diff
@@ -264,28 +289,31 @@ export async function updateChangelog({
       changelog.addRelease({ version: currentVersion });
```

Comment:
`currentVersion` is correctly narrowed due to all related branches being refactored to fall under the `if (isReleaseCandidate)` block instead of there being multiple `if (isReleaseCandidate &&)` blocks. `currentVersion as Version` casting is now unnecessary.